src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkCommunicator.cs

```diff
@@ -0,0 +1,112 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Common.Authentication;
+using Microsoft.Azure.Common.Authentication.Models;
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Azure.Management.Sql;
+using Microsoft.Azure.Management.Sql.Models;
+using Microsoft.WindowsAzure.Management.Storage;
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Commands.Sql.Common;
+
+namespace Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Services
+{
+    /// <summary>
+    /// This class is responsible for all the REST communication with the endpoints
+    /// </summary>
+    public class AzureSqlServerCommunicationLinkCommunicator
+    {
+        /// <summary>
+        /// The Sql client to be used by this end points communicator
+        /// </summary>
+        private static SqlManagementClient SqlClient { get; set; }
+
+        /// <summary>
+        /// Gets or set the Azure subscription
+        /// </summary>
+        private static AzureSubscription Subscription { get; set; }
```

Comment:
It does come from that, it gets passed from AzureSqlServerCommunicationLinkCmdletBase into the adapter and then into the communicator